### PR TITLE
feat: parallelize rows fetcher.

### PR DESCRIPTION
**Pull request by RinChanNOWWW**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

- Make some methods static to support running parallelly.
- Parallelize parquet rows fetcher.
- Parallelize native  rows fetcher.

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/12590)

### CODE REVIEW

File: src/query/storages/fuse/src/operations/read/native_rows_fetcher.rs [hunk 1]
```diff
@@ -80,16 +85,74 @@ impl<const BLOCKING_IO: bool> RowsFetcher for NativeRowsFetcher<BLOCKING_IO> {
...
+        //   Part number of each task   |       Task number
+        // ------------------------------------------------------
+        //    parts_per_thread + 1      |         remain
+        //      parts_per_thread        |   max_threads - remain
+        let num_parts = part_set.len();
+        let parts_per_thread = num_parts / self.max_threads;
+        let remain = num_parts % self.max_threads;
+        let mut tasks = Vec::with_capacity(self.max_threads);
+        // Fetch blocks in parallel.
+        for i in 0..remain {
```
Comment (sundy-li) [inline src/query/storages/fuse/src/operations/read/native_rows_fetcher.rs:116]: duplicated codes with L128-L140.

You can distribute the parts into threads and make the last thread accept the remains.

Comment (sundy-li) [reply src/query/storages/fuse/src/operations/read/native_rows_fetcher.rs:116]: c++ examples:
```
for (size_t thread_idx = 0; thread_idx < num_threads; ++thread_idx)
      {
          size_t begin = num_keys * thread_idx / num_threads;
          size_t end = num_keys * (thread_idx + 1) / num_threads;

           ...
      }
```

File: src/query/storages/fuse/src/operations/read/parquet_rows_fetcher.rs [hunk 2]
```diff
@@ -74,7 +81,67 @@ impl<const BLOCKING_IO: bool> RowsFetcher for ParquetRowsFetcher<BLOCKING_IO> {
...
+        //   Part number of each task   |       Task number
+        // ------------------------------------------------------
+        //    parts_per_thread + 1      |         remain
+        //      parts_per_thread        |   max_threads - remain
+        let num_parts = part_set.len();
+        let parts_per_thread = num_parts / self.max_threads;
+        let remain = num_parts % self.max_threads;
+        let mut tasks = Vec::with_capacity(self.max_threads);
+        // Fetch blocks in parallel.
+        for i in 0..remain {
```
Comment (sundy-li) [inline src/query/storages/fuse/src/operations/read/parquet_rows_fetcher.rs:102]: ditto

**Comment by BohuTANG** [comment]
Cool. It would be great if you could share the benchmarking results if they are ready.

File: src/query/storages/fuse/src/operations/read/native_rows_fetcher.rs (hunk 1 above)
Comment (RinChanNOWWW) [reply src/query/storages/fuse/src/operations/read/native_rows_fetcher.rs:116]: The num_keys will change for different parts.

**Review by sundy-li — APPROVED** [review]

**Comment by RinChanNOWWW** [comment]
```sql
select * from hits order by userid desc limit 1000;
```

Main: 27s
This PR: 1.7s

**Pull request merged**